src/Unused/CLI/Views/SearchResult.hs

```diff
@@ -12,24 +14,24 @@ import qualified Unused.CLI.Views.NoResultsFound as V
 
 searchResults :: [GroupedTerms] -> IO ()
 searchResults terms =
-    printFormattedTerms columnFormat terms
+    runReaderT (printFormattedTerms terms) columnFormat
   where
     allSets = listFromMatchSet =<< map snd terms
     allResults = map snd allSets
     columnFormat = buildColumnFormatter allResults
 
-printFormattedTerms :: ColumnFormat -> [GroupedTerms] -> IO ()
-printFormattedTerms _ [] = V.noResultsFound
-printFormattedTerms cf ts = mapM_ (printGroupingSection cf) ts
+printFormattedTerms :: [GroupedTerms] -> ReaderT ColumnFormat IO ()
```

Comment:
I would probably alias this so `type Search = ReaderT ColumnFormat IO` then abstract your run function:
```
runSearch :: ColumnFormat -> Search ()
runSearch = runReaderT (printFormattedTerms terms)
```